CliFx/Models/TargetCommandSchema.cs

```diff
@@ -0,0 +1,39 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace CliFx.Models
+{
+    /// <summary>
+    /// Defines the target command and the input required for initializing the command.
+    /// </summary>
+    public class TargetCommandSchema
```

Comment:
I guess the name `...Schema` is not accurate in this context because it also encapsulates input. Maybe `CommandCandidate`? I'm open for other names if you have ideas.